miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleController.java

```diff
@@ -178,4 +185,48 @@ public SampleDto getSample(@PathVariable("id") Long id, UriComponentsBuilder uri
     return new ResponseEntity<>(HttpStatus.OK);
   }
 
+  /**
+   * 
+   * @param identitiesSearches
+   *          String externalNames or Identity aliases
+   * @param response
+   * @return
+   * @throws IOException
+   */
+  @RequestMapping(value = "/identities", method = RequestMethod.POST, headers = { "Content-type=application/json" })
+  @ResponseBody
+  public JSONObject getIdentitiesBySearch(@RequestBody JSONObject identitiesSearches,
+      HttpServletResponse response) throws IOException {
+    if (identitiesSearches.size() == 0) {
+      throw new RestException("Must give search terms to look up identities");
+    }
+    Integer requestCounter = (Integer) identitiesSearches.get("requestCounter");
+    JSONArray searchTerms = JSONArray.fromObject(identitiesSearches.get("identitiesSearches"));
+
+    List<Set<SampleDto>> foundIdentities = new ArrayList<Set<SampleDto>>();
+    for (int i = 0; i < searchTerms.size(); i++) {
+      Set<Sample> resultsForOneParent = new HashSet<Sample>();
+      Set<SampleDto> resultsForOneParentDtos = new HashSet<SampleDto>();
+
+      for (String term : IdentityImpl.getSetFromString(((String) searchTerms.get(i)).replaceAll(";", ","))) {
+        Collection<Identity> selectedIdentities = sampleService.getIdentitiesByExternalNameOrAlias(term);
+        if (!selectedIdentities.isEmpty()) {
+          for (Identity selectedIdentity : selectedIdentities) {
+            if (resultsForOneParent.contains(selectedIdentity)) {
+              continue;
+            } else {
+              resultsForOneParentDtos.add(Dtos.asDto(selectedIdentity));
+              resultsForOneParent.add(selectedIdentity);
```

Comment:
No, since `resultsForOneParent` is the set of possibilities for a given externalName string, whereas `foundIdentities` corresponds to the ordered list of given externalNames.